game/hud/src/widgets/Welcome/index.tsx

```diff
@@ -0,0 +1,130 @@
+    // data.message = '<style>' +
+    //   ' .welcome-box { height: 430px; width: 780px; background-image: url("http://i.imgur.com/XLsZQjI.jpg"); background-repeat: no-repeat; background-origin: content-box; }' +
+    //   ' .welcome-msg-bg { position: absolute; top: 0px; left: 10px; height: 406px; width: 780px; background: linear-gradient(rgba(60,50,50,0.5), rgba(60,50,50,0.7)); padding: 0.7em; color: white; }' +
+    //   ' .welcome-msg { position: absolute; top: 2em; left: 2em; height: 77%; width: 94%; background: rgba(255,250,205,0.5); padding: 1em; color: rgba(51,51,0,1); border-radius: 5px; }' +
+    //   '</style>' +
+    //   '<div class="welcome-box"><div class="welcome-msg-bg"><div class="welcome-msg">' +
+    //   '<p>Welcome to the Camelot Unchained Alpha server! Please read this message carefully, as the content is regularly updated.</p>' +
+    //   '<p><b>NOTE:</b> This is an "old-school" Alpha build of the Camelot Unchained client and server. This build is used for testing and demonstration purposes only. This should not be considered a reflection of the final game. Expect stability issues and drastic daily changes/improvements.</p>' +
+    //   '<p><u>Known Issues:</u><br /> - Animations are not final. They will be improved!<br /> - Movement speeds are not final.<br /> - Abilities are disabled.</p>' +
+    //   '<p>To view the Alpha Manual: <a href="https://forums.camelotunchained.com/topic/10538-alpha-manual-v042-updated-27-july-2015/" target="_new">CLICK HERE</a><br />' +
+    //   'To report bugs: <a href="https://forums.camelotunchained.com/forum/212-current-bugs-extended-alpha/" target="_new">CLICK HERE</a><br />' +
+    //   'To provide feedback/suggestions: <a href="https://forums.camelotunchained.com/forum/194-alpha-feedback-suggestions/" target="_new">CLICK HERE</a></p>' +
+    //   '<p>Try pressing <b style="color: rgba(110,45,23,1); font-size: 125%;">V</b> to create new abilities and <b style="color: rgba(110,45,23,1); font-size: 125%;">B</b> to open your ability book. Clicking the lock icon (lower-right corner) allows resizing and moving UI widgets. ' +
+    //   'For some helpful backer-supplied tips, type <b style="color: rgba(110,45,23,1); font-size: 125%;"><u>!tips</u></b> in the chat window.</p>' +
+    //   '</div></div></div>';
+
+    welcomeMessage.push(
+      <div key='100' dangerouslySetInnerHTML={{__html: data.message}} />
+    );
+    this.setState({ message: welcomeMessage } as any);
+  }
+
+  onMessageFailed = (error: any) => {
+    //**TODO: Proper logging?
+    console.log(error);
+  }
+
+  renderButtons = (key: number): JSX.Element[] => {
+    const buttons: JSX.Element[] = [];
+
+    buttons.push(
+      <a key={++key} className='btn cu-btn-dismiss' onClick={this.hide}>Dismiss</a>
+    );
+    buttons.push(
+      <a key={++key} className='cu-btn-dismiss-delay' onClick={this.hideDelay}>Dismiss For 24h</a>
+    );
+    return buttons;
```

Comment:
I don't see why this exists.  This is just extra work when you can simply add it inline in the JSX below.

```
<div className='cu-btnbar'>
  <a key={++key} className='btn cu-btn-dismiss' onClick={this.hide}>Dismiss</a>
  <a key={++key} className='cu-btn-dismiss-delay' onClick={this.hideDelay}>Dismiss For 24h</a>
</div>
```